TST, MAINT: test_maxiter_worsening on Python 3.9 + Linux ARM64

`test_maxiter_worsening[gmres]` was observed to fail in the [wheels repo CI](https://github.com/MacPython/scipy-wheels/pull/98) for Python 3.9 + Linux ARM64: https://travis-ci.com/github/MacPython/scipy-wheels/jobs/422619853 

That's the only remaining failure I see before we're in good shape for 3.9 wheels across the board. I'm going to go ahead and restart that job now, but I'll paste the summary error output below in case I'm not lucky with the restart:

```
_______________________ test_maxiter_worsening[gmres] _________________________

[gw5] linux -- Python 3.9.0 /venv/bin/python

/venv/lib/python3.9/site-packages/scipy/sparse/linalg/isolve/tests/test_iterative.py:482: in test_maxiter_worsening

    assert_(error <= tol*best_error)

E   AssertionError

        A          = array([[-0.11127953+0.j        ,  0.        +0.j        ,

         0.        +0.j        ,  0.16127953+0.j        ],

 ...,

       [ 0.11127953+0.j        ,  0.        +0.j        ,

         0.        +0.j        , -0.16127953+0.j        ]])

        best_error = 0.001979511632724729

        error      = 0.05633679479248356

        info       = 7

        maxiter    = 7

        solver     = <function gmres at 0xffff74623040>

        tol        = 7

        v          = array([1., 1., 1., 1.])

        x          = array([-1.18884429e+15+3.26937190e+09j, -3.44234853e-03-1.59087294e-01j,

       -3.44394082e-03+1.59079391e-01j, -8.20277894e+14+2.25579878e+09j])
```

--- a/dev.py
+++ b/dev.py
@@ -309,11 +309,6 @@ class Dirs:
         else:
             self.installed = self.build.parent / (self.build.stem + "-install")
 
-        if sys.platform == 'win32' and sys.version_info < (3, 10):
-            # Work around a pathlib bug; these must be absolute paths
-            self.build = Path(os.path.abspath(self.build))
-            self.installed = Path(os.path.abspath(self.installed))
-
         # relative path for site-package with py version
         # i.e. 'lib/python3.10/site-packages'
         self.site = self.get_site_packages()

--- a/scipy/sparse/linalg/_isolve/tests/test_iterative.py
+++ b/scipy/sparse/linalg/_isolve/tests/test_iterative.py
@@ -3,7 +3,6 @@
 
 import itertools
 import platform
-import sys
 import pytest
 
 import numpy as np
@@ -451,9 +450,6 @@ def test_maxiter_worsening(solver):
     # This can occur due to the solvers hitting close to breakdown,
     # which they should detect and halt as necessary.
     # cf. gh-9100
-    if (solver is gmres and platform.machine() == 'aarch64'
-            and sys.version_info[1] == 9):
-        pytest.xfail(reason="gh-13019")
     if (solver is lgmres and
             platform.machine() not in ['x86_64' 'x86', 'aarch64', 'arm64']):
         # see gh-17839

